boards/arm/adafruit_kb2040/adafruit_kb2040.dts

```diff
@@ -95,6 +98,25 @@
 	pinctrl-names = "default";
 };
 
+&pio0 {
+	status = "okay";
+	pinctrl-0 = <&ws2812_pio0_default>;
+	pinctrl-names = "default";
+	state-machine-num = <1>;
+
+	gpios = <&gpio0 17 0>;
```

Comment:
My problem with this is that it requires you to redefine the pin here (as 17), when you've already done so in the pinctrl state. Now to change the pin you have to change both. Ideally, pinctrl would allow us to define names for pins in the pinctrl state, but currently it isn't supported. Any other ideas?

Reply:
IMO, there isn't a better way at this point.
Of course, I also considered using pinctrl, but there were some difficulties.

First, pinctrl has a hierarchical structure.
Creating a macro that statically resolves name-to-pin correspondences takes a lot of work.
(It should change dts parsing script.)

We also need to determine the valid scope definitions for names.
Also, pinctrl can switch between sets of settings depending on the state. (Only default can select for rpi_pico at this time.) 
Since the changed setting may not always include the definition of the pin name, it is necessary to consider the behavior at that time.

Given the above, this is out of the current pinctrl design scope, so I'm choosing the current method.
It would be a good idea to add pin names in pinctrl, but it's a too big issue to address in this PR.